### DISCUSSION TITLE
Keep the case of file names in URLs

### DISCUSSION THREAD

**Discussion by ybw0014** [withastro/starlight#1268] [Feature Requests]
### What version of `starlight` are you using?

0.15.0

### What is your idea?

have an option to keep the case of files instead of all lowercase.
or, lowercase the request URL internally so starlight can still show the content and not throw 404.

### Why is this feature necessary?

I would like to migrate to Starlight, but the old docs URLs are case-sensitive. The page names are in this format `Some-Pages`. I do not want the users to see 404 pages when they visit with the same URL.

### Participation

- [ ] I am willing to submit a pull request for this feature.

**Comment by HiDeoo** [comment]
Thanks for your feedback.

Starting with [Starlight 0.35.0](https://github.com/withastro/starlight/releases/tag/%40astrojs%2Fstarlight%400.35.0), a new [`generateId` option](https://starlight.astro.build/reference/configuration/#generateid) was added to Starlight’s [`docsLoader()`](https://starlight.astro.build/reference/configuration/#docsloader) to allow for such customizations.

For example, the following code snippet will remove the `.md` or `.mdx` extension from the filenames, but otherwise leave them unchanged:

```js
docsLoader({
  // Remove the `.md` or `.mdx` extension, but otherwise don’t process filenames.
  generateId: ({ entry }) => entry.split('.').slice(0, -1).join('.'),
}),
```
